pixl_core/src/core/uploader/_dicomweb.py

```diff
@@ -73,45 +81,69 @@ def send_via_stow(self, resource_id: str) -> requests.Response:
         return response
 
-    def _check_dicomweb_server(self) -> bool:
+    def _check_dicomweb_server_exists(self) -> bool:
         """Checks if the dicomweb server exists."""
-        response = requests.get(self.url, auth=(self.user, self.password), timeout=30)
+        response = requests.get(
+            self.orthanc_dicomweb_url,
+            auth=(self.orthanc_user, self.orthanc_password),
+            timeout=self.http_timeout,
+        )
         success_code = 200
         if response.status_code != success_code:
             return False
         return True
 
+    def _validate_dicomweb_server(self) -> None:
+        """Check if the DICOMweb server is reachable from within the Orthanc instance."""
+        connection_error = requests.exceptions.ConnectionError("DICOMweb server not reachable")
+        if not self._check_dicomweb_server_exists():
+            logger.error("DICOMweb server not configured")
+            raise connection_error
+
+        # If dicomweb server configured, check if we can reach it
+        try:
+            response = requests.post(
+                self.orthanc_dicomweb_url + "/get",
+                auth=(self.orthanc_user, self.orthanc_password),
+                data='{"Uri": "/studies"}',
+                headers={"content-type": "application/x-www-form-urlencoded"},
+                timeout=self.http_timeout,
+            )
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            logger.error("Failed to reach DICOMweb server")
+            raise connection_error from e
+
     def _setup_dicomweb_credentials(self) -> None:
         """
         Add the necessary credentials to the DicomWeb server in Orthanc.
-        This dyniamically creates a new endpoint in Orthanc with the necessary credentials, so we
+        This dynamically creates a new endpoint in Orthanc with the necessary credentials, so we
         can avoid hardcoding the credentials in the Orthanc configuration at build time.
         """
-        DICOM_ENDPOINT_URL = config("DICOM_ENDPOINT_URL")
-        HTTP_TIMEOUT = int(config("HTTP_TIMEOUT", default=30))
-
         dicomweb_config = {
-            "Url": DICOM_ENDPOINT_URL,
-            "Username": self.user,
-            "Password": self.password,
+            "Url": self.endpoint_url,
+            "Username": self.endpoint_user,
+            "Password": self.endpoint_password,
             "HasDelete": True,
-            "Timeout": HTTP_TIMEOUT,
+            "Timeout": self.http_timeout,
         }
 
         headers = {"content-type": "application/json"}
         try:
             requests.put(
-                self.url,
-                auth=(self.user, self.password),
+                self.orthanc_dicomweb_url,
+                auth=(self.orthanc_user, self.orthanc_password),
                 headers=headers,
                 data=json.dumps(dicomweb_config),
                 timeout=10,
             )
```

Comment:
```suggestion
            )
            response.raise_for_status()
```
I spotted another one

Reply:
😅 Thanks Jeremy!